dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/EventChangeLogServiceTest.java

```diff
@@ -463,9 +459,225 @@ void shouldReturnEventPropertiesChangeLogWhenExistingGeometryPointPropertyDelete
+  private static Stream<Arguments> provideOrderParamsDesc() {
+    return Stream.of(
+        Arguments.of(
+            "createdAt",
+            SortDirection.DESC,
+            Arrays.asList("25", "20", "15"),
+            Arrays.asList("20", "15", null)),
+        Arguments.of(
+            "username",
+            SortDirection.DESC,
+            Arrays.asList("25", "20", "15"),
+            Arrays.asList("20", "15", null)),
+        Arguments.of(
+            "username",
+            SortDirection.ASC,
+            Arrays.asList("25", "20", "15"),
+            Arrays.asList("20", "15", null)));
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideOrderParamsDesc")
+  void shouldSortChangeLogsByCreatedAtDescWhenOrderingByDateOrUsername(
+      String field,
+      SortDirection sortDirection,
+      List<String> currentValues,
+      List<String> previousValues)
+      throws ForbiddenException, NotFoundException {
+    EventChangeLogOperationParams params =
+        EventChangeLogOperationParams.builder().orderBy(field, sortDirection).build();
+    Event event = getEvent("QRYjLTiJTrA");
+    String dataElementUid = event.getEventDataValues().iterator().next().getDataElement();
+
+    updateDataValues(event, dataElementUid, "20", "25");
+
+    List<EventChangeLog> changeLogs =
+        getDataElementChangeLogs(
+            eventChangeLogService.getEventChangeLog(
+                UID.of("QRYjLTiJTrA"), params, defaultPageParams));
+
+    assertNumberOfChanges(3, changeLogs);
+    assertAll(
+        () ->
+            assertDataElementUpdate(
+                "GieVkTxp4HH", previousValues.get(0), currentValues.get(0), changeLogs.get(0)),
+        () ->
+            assertDataElementUpdate(
+                "GieVkTxp4HH", previousValues.get(1), currentValues.get(1), changeLogs.get(1)),
+        () -> assertDataElementCreate("GieVkTxp4HH", currentValues.get(2), changeLogs.get(2)));
+  }
+
+  @Test
+  void shouldSortChangeLogsWhenOrderingByCreatedAtAsc()
+      throws ForbiddenException, NotFoundException {
+
+    EventChangeLogOperationParams params =
+        EventChangeLogOperationParams.builder().orderBy("createdAt", SortDirection.ASC).build();
+
+    Event event = getEvent("QRYjLTiJTrA");
+    String dataElementUid = event.getEventDataValues().iterator().next().getDataElement();
+
+    updateDataValues(event, dataElementUid, "20", "25");
+
+    List<EventChangeLog> changeLogs =
+        getDataElementChangeLogs(
+            eventChangeLogService.getEventChangeLog(
+                UID.of("QRYjLTiJTrA"), params, defaultPageParams));
+
+    assertNumberOfChanges(3, changeLogs);
+    assertAll(
+        () -> assertDataElementCreate(dataElementUid, "15", changeLogs.get(0)),
+        () -> assertDataElementUpdate(dataElementUid, "15", "20", changeLogs.get(1)),
+        () -> assertDataElementUpdate(dataElementUid, "20", "25", changeLogs.get(2)));
+  }
+
+  @Test
+  void shouldSortChangeLogsWhenOrderingByCreatedAtDesc()
```

Comment:
is that not already part of `shouldSortChangeLogsByCreatedAtDescWhenOrderingByDateOrUsername` element 1 🤔